### PR TITLE
[core] Remove serialized_allocated_resource_instances from RuntimeEnv creation.

### PR DESCRIPTION
**Pull request by rynewang**
GetOrCreateRuntimeEnvRequest has field `serialized_allocated_resource_instances`. It does not serve any purpose other than a log in the Runtime Env Agent. Removes it.

Also removes required_resources in WorkerCacheKey and the flag worker_resource_limits_enabled.

### CODE REVIEW

**Comment by jjyao** [comment]
Past conversation on deleting the code: https://github.com/ray-project/ray/issues/16871#issuecomment-1439566066

**Comment by jjyao** [comment]
I'd vote for removing `worker_resource_limits_enabled` flag completely for now and add it back later once we actually support it.

Also I feel the implementation can be changed to dynamically inject the allocated resources into container runtime env dict (e.g.  `{"container": {"image": xxxx, "allocated_resources": {"CPU": 1}}}`) before calling `PopWorker` instead of passing it separately. cc @SongGuyang.

**Review by rkooo567 — COMMENTED** [review]
Let's also remove worker_resource_limits_enabled?

**Comment by rynewang** [comment]
what is allocated resources btw?

**Comment by jjyao** [comment]
> what is allocated resources btw?

That's allocated resources to the task.

**Comment by rynewang** [comment]
worker_resource_limits_enabled removed

File: src/ray/common/task/task_spec.cc [hunk 1]
```diff
@@ -594,15 +594,13 @@ std::string TaskSpecification::CallSiteString() const {
...
     const absl::flat_hash_map<std::string, double> &required_resources,
     bool is_actor,
     bool is_gpu,
     bool is_root_detached_actor)
-    : serialized_runtime_env(serialized_runtime_env),
-      required_resources(RayConfig::instance().worker_resource_limits_enabled()
-                             ? required_resources
-                             : absl::flat_hash_map<std::string, double>{}),
+    : serialized_runtime_env(std::move(serialized_runtime_env)),
+      required_resources(required_resources),
```
Comment (jjyao) [inline src/ray/common/task/task_spec.cc:603]: Required_resources can also be removed?

Comment (rynewang) [reply src/ray/common/task/task_spec.cc:603]: I don't think it's always empty. It describes the resource shape of the task, populated from Python side to CoreWorker: https://github.com/ray-project/ray/blob/e208dc550cd5705115c13e4dd8a5abcfc5ff3900/src/ray/core_worker/core_worker.cc#L2552

Comment (jjyao) [reply src/ray/common/task/task_spec.cc:603]: I mean `required_resources` of `WorkerCacheKey` is always empty if the flag is off?

**Review by rkooo567 — APPROVED** [review]
LGTM. I will leave required_resources discussion to @jjyao

File: src/ray/common/task/task_spec.cc (hunk 1 above)
Comment (rynewang) [reply src/ray/common/task/task_spec.cc]: removed

**Review by jjyao — APPROVED** [review]
nice

**Pull request merged**